src/common/drain.rs

```diff
@@ -1,7 +1,9 @@
 use std::mem;
 
 use pin_project::pin_project;
-use tokio::sync::{mpsc, watch};
+use tokio::stream::Stream;
+use tokio::sync::mpsc;
+use tokio02::sync::watch;
```

Comment:
:(

I can't remember, is the way forward here to change so just holding a `Pin<Box<dyn Future>>`, that we build by using the async fn in a loop until draining is signaled?

Reply:
Unfortunately, the current `watch` API in tokio 0.3 doesn't provide a (safe) way of doing that, since the `changed()` function borrows the `Watch` (https://docs.rs/tokio/0.3.3/tokio/sync/watch/struct.Receiver.html#method.changed).

Reply:
fixed! 7fd643a